bin/pycbc_live

```diff
@@ -605,17 +646,33 @@ class LiveEventManager(object):
                 self.ifar_upload_threshold < single['foreground/ifar'] and \
                 not any(nearby_coincs) and \
                 self.run_snr_optimization
+
+            # Save the event
             if not upload_checks:
                 event.save(fname)
+
+            # Save the event info/data and what _would_ be run for SNR optimizer,
+            # even if not running it - do this before the thread so no
+            # data buffers move on in a possible interim period
+
+            # Which IFOs were active?
+            live_ifos = [ifo for ifo in sld if 'snr_series' in sld[ifo]]
+
+            # Tell SNR optimized event about p_terr
+            if hasattr(event, 'p_terr') and event.p_terr is not None:
+                single['p_terr'] = event.p_terr
+
+            # Do the optimizer setup
+            cmd, out_dir_path = self.setup_optimize_snr(
+                single,
+                live_ifos,
+                [ifo],
+                fname,
+            )
```

Comment:
This will run even if there was a nearby coinc, and will generate a bunch of files as a consequence, which does not sound like a desirable behavior. Maybe this call should go into the `if upload_checks or optimize_snr_checks` condition below? And I guess the same should happen for the similar code above (watch out for code duplication…)

Reply:
I.e. what is the use case for doing anything if we are neither uploading nor running the SNR optimizer?

Reply:
Hold on, I am conflating two different decisions in my two comments above. One decision is what to do in terms of SNR optimization of singles that have a nearby coinc. The other decision is what to do in terms of generating SNR optimization input if we are neither uploading nor SNR-optimizing.

Reply:
Ok, I think we generally do want to call `setup_optimize_snr` even if `upload_checks` and `optimize_snr_checks` are both false: this would allow us to generate some files for testing the SNR optimizer later on.

However, if there is a nearby coinc, then `upload_checks` and `optimize_snr_checks` will both be false, and I argue that we should *not* call `setup_optimize_snr`.

Reply:
I have added in a `if nearby_coincs: continue` statement so that this bit isn't reached in that case.

Reply:
I don't see the "similar code above" though, I think that setup_optimize_snr should be called for _all_ coincs, right?

Reply:
Yes nevermind, that was a thinko.